pkg/kubectl/cmd/annotate.go

```diff
@@ -193,7 +204,11 @@ func (o AnnotateOptions) RunAnnotate() error {
 			return err
 		}
 
-		name, namespace, obj := info.Name, info.Namespace, info.Object
+		obj, err := resource.AsVersionedObject([]*resource.Info{info}, false, o.outputVersion.String())
```

Comment:
I would have expected you to use the version you negotiated with the API server so that when you submit the patch, your patch matches the version you're writing to.